File: Python/0051-0100/66. Plus One.py
# https://leetcode.com/problems/plus-one/

class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        size = len(digits)
        rem = 1

        for index in reversed(range(0,size)):
            if rem == 1:
                digits[index] += 1
                rem = 0 if (digits[index] <= 9) else 1
                digits[index] = digits[index] % 10

        if rem == 1:
            digits.insert(0, 1)
        
        return digits

File: Java/0051-0100/66. Plus One.java
// https://leetcode.com/problems/plus-one/

class Solution {
    public int[] plusOne(int[] digits) {
        int size = digits.length;
        int[] result = new int[size+1];
        int rem = 1;

        for(int index=size-1; index>=0; index--){
            if(rem == 1){
                digits[index]++;
                rem = (digits[index] <= 9) ? 0 : 1;
                digits[index] = digits[index]%10;
            }
            result[index+1] = digits[index];
        }

        if(rem == 0)
            return digits;
        
        result[0] = 1;
        return result;
    }
}
